packages/components/atoms/f-segmented-control/src/components/SegmentedControl.vue

```diff
@@ -0,0 +1,254 @@
+<template>
+    <div
+        :class="{
+            [$style['c-segmented-control']]: true,
+            [$style['c-segmented-control--large']]: size === 'large'
+        }"
+        data-test-id="segmented-control"
+        :aria-label="screenreaderLabel"
+        role="radiogroup"
+        tabindex="-1"
+        @keydown="onKeyDown"
+    >
+        <button
+            v-for="(option, index) in options"
+            :key="option.label"
+            :data-test-id="`segmented-control-option-${index}`"
+            type="button"
+            :class="{
+                [$style['c-segmented-control__option']]: true,
+                [$style['c-segmented-control__option--selected']]: option.label === selectedLabel
+            }"
+            :aria-checked="option.label === selectedLabel ? 'true' : 'false'"
+            role="radio"
+            :tabindex="focusedTabIndex[index]"
+            :disabled="option.disabled"
+            @click="selectOption(option.label)"
+        >
+            <!-- TODO: This is a placeholder for adding vue icons -->
+            <span>{{ option.label }}</span>
+        </button>
+    </div>
+</template>
+
+<script>
+export default {
+    name: 'SegmentedControl',
+    props: {
+        screenreaderLabel: {
+            type: String,
+            required: true
+        },
+        options: {
+            type: Array,
+            required: true,
+            validator: value => {
+                if (value.length < 2 || value.length > 4) {
```

Comment:
Could we move this to an external file and import it?

Reply:
I would split these validations up into functions and then you could call them (and test them) individually. i.e.

```js
validator: value => {
   isValid = isValidNumberOfOptions(value) && areOptionTypesValid(value);
   return isValid;
}